source/docs/advanced/production-ready/media-middleware.md

```diff
@@ -117,6 +117,105 @@ For a more data-efficient browser preload, we also recommend that you define the
+2. Open the developers tools and paste the below snippet in the `console` tab.
+3. Before you hit the enter key edit the condition indicated in the snippet to match the image you want.
+4. Hit the enter key.
+5. Now you have 4 new global functions you can use `getImage`, `resetIsMyImage`, `getSizesStats`, `clearSizesStats`.
+   - Use the `getImage` function to test the condition you set if it returns the correct image.
+   - In case `getImage` was returning the wrong image. Use `resetIsMyImage` to change the condition.
+   - `getSizesStats` returns the sizes stats collected so far.
+   - `clearSizesStats` clears the stats collected so far.
+6. To start collecting stats of how your image width changes with viewport width start resizing your browser window from small (say 300px) to extra large. Be gentle with the resizing so as to capture more data points. Be extra gentle around break points to capture the breakpoint perfectly.
+7. Run `getSizesStats()` in your `console` tab. It will print a long string.
+8. Copy the string in 7. (everything in between the double quotes).
+9. Paste the string you copied in 8. In to a spreadsheet.
+10. Now you can plot how the image width changes with viewport width.
+11. Using the above information and the different sizes available for your image, you can build a `sizes` value that matches your scenario. Check [example below](#image-sizes-example) for a hands-on exercise.
+
+```js
+var { getSizesStats, getImage, resetIsMyImage, clearSizesStats } = ((
+  isMyImage = (img) => {
+    return (
+      // IMPORTANT UPDATE THE CONDITION BELOW TO MATCH THE IMAGE YOU WANT TO TRACK
+      (img.alt || "").trim().toLowerCase() ===
+        "your-image-alt".trim().toLowerCase() &&
+      (img.src || "").trim().toLowerCase().indexOf("your-image-src".trim().toLowerCase()) >= 0 &&
+      (img?.attributes?.some_custom_prop?.value || "").trim().toLowerCase() ===
+        "your-custom-image-prop-value".trim().toLowerCase() &&
+      (img.className || "").toLowerCase().indexOf("your-class-name") >= 0
+    );
+  }
+) => {
+  const getImage = () => {
+    return Array.prototype.filter.call(
+      document.getElementsByTagName("img"),
+      isMyImage
+    )[0];
+  };
+
+  const stats = [];
+  window.addEventListener("resize", () => {
+    const windowSize = window.innerWidth;
+    const img = getImage();
+    if (!stats.find(([winSize]) => winSize === windowSize)) {
+      stats.push([windowSize, img.offsetWidth]);
+    }
+  });
+  return {
+    getSizesStats: () => {
+      stats.sort(([winSize1], [winSize2]) => winSize1 - winSize2);
+      return stats.map((itm) => itm.join("\t")).join("\n");
+    },
+    getImage,
+    clearSizesStats: () => {
+      stats.splice(0, stats.length);
+    },
+    resetIsMyImage: (newIsMyImage) => {
+      isMyImage = newIsMyImage;
+    },
+  };
+})();
+```
+
+### Image Sizes Example:
+
+Let's say the data you collected in the [image sizes section above](#image-sizes) are as follows:
```

Comment:
This anchor doesn't work. It should be fixed if you accept the above suggestion about a new heading.